swoole_http_client_coro.cc

```
@@ -1404,6 +1413,30 @@ bool http_client::close()
     return false;
 }
 
+std::string http_client::get_basic_auth_encode(char *name, size_t nameLen, char *passwd, size_t passwdLen){
+    size_t inputLen = nameLen + passwdLen + 1;
+    size_t outputLen = BASE64_ENCODE_OUT_SIZE(inputLen);
+    char *input = new char[inputLen + 1];
+    char *output = new char[outputLen + 7];
+    if(input == nullptr || output == nullptr) return std::string();
```

C++ will throw an exception instead of return nullptr